tools/ci.sh

```diff
@@ -29,6 +29,30 @@ function ci_code_formatting_run {
     tools/codeformat.py -v
 }
 
+########################################################################################
+# commit formatting
+
+function ci_commit_format_check {
+    # read from pipe
+    read sha
+    # test subject matches format we want
+    msg=$(git log -1 "$sha" --pretty=format:"%s")
+    echo $msg | grep -q -E '^.+: [A-Z]+.+ .+\.$'
+    if [ $? -ne 0 ]; then
+        echo "Incorrect commit subject line format: '$msg'"
+        exit 1
+    fi
+    # test character limit on all lines of complete message
+    git log -1 $sha --pretty=format:"%s%n%b" |
+        awk '{if(length > 75){print "Line > 75 characters: " $0; exit 1}}'
+}
+
+function ci_commit_formatting_run {
+    git remote add upstream https://github.com/micropython/micropython.git
+    git fetch --depth=100 upstream
```

Comment:
The checked out repo by default is cloned with depth 1 though, i.e. only the last commit. Morever we want to compare the last commit the against the latest master of the main repo (so not the one where this might be built) because that's eventually where the PR would end up. See also code size check, it does the same. This was discussed at one point but I cannot find where anymore :(